Remove documentation on TimeDeltaSensorAsync

Subham-KRLX opened the pull request:
<!--
 Licensed to the Apache Software Foundation (ASF) under one
 or more contributor license agreements.  See the NOTICE file
 distributed with this work for additional information
 regarding copyright ownership.  The ASF licenses this file
 to you under the Apache License, Version 2.0 (the
 "License"); you may not use this file except in compliance
 with the License.  You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

 Unless required by applicable law or agreed to in writing,
 software distributed under the License is distributed on an
 "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 KIND, either express or implied.  See the License for the
 specific language governing permissions and limitations
 under the License.
 -->

<!--
Thank you for contributing! Please make sure that your code changes
are covered with tests. And in case of new features or big changes
remember to adjust the documentation.

Feel free to ping committers for the review!

In case of an existing issue, reference it using one of the following:

closes: #ISSUE
related: #ISSUE

How to write a good git commit message:
http://chris.beams.io/posts/git-commit/
-->

Remove deprecated TimeDeltaSensorAsync class

Removes `TimeDeltaSensorAsync` class which was deprecated in favor of the standard [TimeDeltaSensor(deferrable=True)](cci:2://file:///Users/subhamsangwan/airflow/providers/standard/src/airflow/providers/standard/sensors/time_delta.py:41:0-144:19) pattern used by other Airflow sensors.

**Changes:**
- Removed `TimeDeltaSensorAsync` class and exports
- Updated tests and documentation to use [TimeDeltaSensor(deferrable=True)](cci:2://file:///Users/subhamsangwan/airflow/providers/standard/src/airflow/providers/standard/sensors/time_delta.py:41:0-144:19)

closes: #59644

<!-- Please keep an empty line above the dashes. -->
---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [airflow-core/newsfragments](https://github.com/apache/airflow/tree/main/airflow-core/newsfragments).

fritz-astronomer commented on airflow-core/tests/unit/models/test_taskinstance.py:
@Subham-KRLX - are you able to rebase without 
https://github.com/apache/airflow/pull/58594/
changes?

(unless they're also helpful here, and, forgive if so)

fritz-astronomer commented on PR_COMMENT.md:
`rm`

fritz-astronomer commented on PR_COMMENT_FINAL.md:
`rm`

fritz-astronomer commented on PR_RESPONSE.md:
`rm`

fritz-astronomer commented on PR_DESCRIPTION.md:
`rm`

fritz-astronomer commented on providers/standard/docs/sensors/datetime.rst:
```diff
@@ -23,30 +23,43 @@ TimeDeltaSensor
-It is an async version of the operator and requires Triggerer to run.
+* Frees worker slots during waiting periods
+* More efficient resource utilization for long waits
+* Ideal for production environments with limited worker capacity
```
I'd instead point to https://airflow.apache.org/docs/apache-airflow/stable/authoring-and-scheduling/deferring.html rather than reiterating benefits

fritz-astronomer commented on providers/standard/tests/unit/standard/sensors/test_time_delta.py:
```diff
@@ -172,8 +172,9 @@ def setup_method(self):
         defer_mock = mocker.patch(DEFER_PATH)
         delta = timedelta(hours=1)
-        with pytest.warns(AirflowProviderDeprecationWarning):
-            op = TimeDeltaSensorAsync(task_id="timedelta_sensor_check", delta=delta, dag=self.dag)
```
we might still want the deprecation warning? And not to fully remove the sensor - the deprecation said next **_major version_**

Subham-KRLX committed:
docs: improve TimeDeltaSensor documentation with deferrable mode reference

- Remove separate TimeDeltaSensorAsync documentation section
- Add reference to official deferring documentation
- Show both sync and deferrable examples using TimeDeltaSensor

Related: #59644

Subham-KRLX committed:
docs: remove TimeDeltaSensorAsync documentation

jroachgolf84 reviewed (approved):
I'm not a committer, but this looks good to me. Thanks for doing this!

uranusjr committed:
Merge branch 'main' into remove-timedelta-sensor-async

amoghrajesh committed:
Merge branch 'main' into remove-timedelta-sensor-async

Subham-KRLX committed:
Merge branch 'main' into remove-timedelta-sensor-async
